lib/rules/selector-max-specificity/index.mjs

```diff
@@ -234,35 +232,27 @@ const rule = (primary, secondaryOptions) => {
 		const maxSpecificity = { a, b, c };
 
 		root.walkRules((ruleNode) => {
-			if (!isStandardSyntaxRule(ruleNode)) {
-				return;
-			}
-
-			// Using `.selectors` gets us each selector in the eventuality we have a comma separated set
-			for (const selector of ruleNode.selectors) {
-				for (const resolvedSelector of resolvedNestedSelector(selector, ruleNode)) {
-					// Skip non-standard syntax selectors
-					if (!isStandardSyntaxSelector(resolvedSelector)) {
-						continue;
-					}
-
-					const selectorTree = parseSelector(resolvedSelector, result, ruleNode);
-
-					if (!selectorTree) continue;
+			if (!isStandardSyntaxRule(ruleNode)) return;
 
+			flattenNestedSelectorsForRule(ruleNode, result).forEach(({ selector, resolvedSelectors }) => {
+				resolvedSelectors.forEach((resolvedSelector) => {
 					// Check if the selector specificity exceeds the allowed maximum
-					if (compare(maxChildSpecificity(selectorTree), maxSpecificity) > 0) {
+					if (compare(nodeSpecificity(resolvedSelector), maxSpecificity) > 0) {
+						const index = selector.first?.sourceIndex ?? 0;
```

Comment:
[note] The coverage diff is caused by this line, but I think it's ignorable.

If improving the coverage, we might change it like this, but I'm unsure it's really safe...

```suggestion
						const index = selector.first.sourceIndex;
```

Reply:
Yeah, I think it is safe in practice, but I'd rather keep it anyway.

If the AST is mutated without setting `sourceIndex` or if something clears out a selector it would throw errors.

But maybe we can add a helper so that we can document this better?

Reply:
I am fine with keeping it as-is, and I agree with adding a helper. 👍🏼

Reply:
I should actually fix this one : https://github.com/postcss/postcss-selector-parser/issues/287

Then we change `const index = selector.first?.sourceIndex ?? 0` to `const index = selector.sourceIndex`

Reply:
nice👍